pkg/cmd/cli/cmd/process.go

```diff
@@ -126,12 +127,20 @@ func NewCmdProcess(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.
 
 			// If 'parameters' flag is set it does not do processing but only print
 			// the template parameters to console for inspection.
-			if cmdutil.GetFlagBool(cmd, "parameters") == true {
+			if kcmdutil.GetFlagBool(cmd, "parameters") == true {
 				err = describe.PrintTemplateParameters(templateObj.Parameters, out)
 				checkErr(err)
 				return
 			}
 
+			label := kcmdutil.GetFlagString(cmd, "label")
+			if len(label) == 0 {
+				// Auto-generate a label
+			}
+			lbl, err := cmdutil.ParseLabel(label)
+			checkErr(err)
+			template.AddObjectLabels(templateObj, lbl)
```

Comment:
@bparees is this enough? Will all the generated resources from this template object have that `label`?